kqueue.c

```diff
@@ -210,6 +211,12 @@ kq_build_changes_list(const struct event_changelist *changelist,
 			int newsize = kqop->changes_size * 2;
 			struct kevent *newchanges;
 
+			if (newsize < 0 || (size_t)newsize >
```

Comment:
`newsize < 0` does not work to detect overflow, because the compiler is likely to assume that this does not happen because signed int overflow is undefined, and it may remove this check completely as an optimization.

What might work is `kqop->changes_size > INT_MAX / 2` ... see https://stackoverflow.com/a/25151704

Reply:
Indeed, that could be optimized away. Maybe the cast to size_t could still trigger the second condition, but I will create a pull request to avoid undefined behavious.

Reply:
>because signed int overflow is undefined

Sigh. I was thinking about moving that check before assigning `newsize` but after jumping back and forward forgot, thanks!

Reply:
Fixed in https://github.com/libevent/libevent/pull/813